Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/Search/Filter.cpp

```diff
@@ -362,6 +362,13 @@ namespace AzToolsFramework
 
         bool AssetGroupFilter::MatchInternal(const AssetBrowserEntry* entry) const
         {
+            AZStd::string ext;
+            AzFramework::StringFunc::Path::GetExtension(entry->GetVisiblePath().c_str(), ext, false);
+            if (m_group == "Shader" && ext.starts_with("shadervariant"))
+            {
+                return true;
+            }
```

Comment:
Here is a hardcoded change that results in the desired effect. To properly implement this behavior, we have a problem because the filter system is now checking the AssetGroup static string member which only exists on product assets, but the `shaderviarant` is a source.

Reply:
See AtomToolsAssetBrowser::CreateFilter() const for how the atom tools set up the asset browser filter. I created a custom filter type that just uses a standard function/lambda. The default function excludes files in the cache folder. You can update the class to pass in a custom filter function to the constructor that includes the cache folder and your specific extensions. Try to preserve the default behavior and verify in the material editor and material canvas.

Reply:
I attempted a change here to let this `GroupAssetFilter` pass sources asset. As it was, it only tolerated `Products`. Which is odd because of how you setup the root filter (with an `AND` policy) in `..::CreateFilter()` where it does the opposite, products asset can't pass the `!=Source !=Folder` test. According to first order logic, when an asset group filter is set, nothing should be displayed ever. But it's not what we observe. Anyways, since .shadervariantlist is a source asset, and that's what we're interested-in, in the SMC anway, I added a possible implementation to let source asset pass that filter. Not sure about it though, because it uses a fuzzy "get primary asset type".